x/merkledb/db.go

```diff
@@ -1213,10 +1225,10 @@ func getBufferFromPool(bufferPool *sync.Pool, size int) []byte {
 }
 
 // cacheEntrySize returns a rough approximation of the memory consumed by storing the path and node
-func cacheEntrySize(p path, n *node) int {
+func cacheEntrySize(p Path, n *node) int {
 	if n == nil {
-		return len(p)
+		return len(p.Bytes())
 	}
 	// nodes cache their bytes representation so the total memory consumed is roughly twice that
-	return len(p) + 2*len(n.bytes())
+	return len(p.Bytes()) + 2*len(n.bytes())
```

Comment:
should 2 depend on the branch factor here?

Reply:
No, this 2 is a reference to the fact that nodes store a copy of their byte representation in them, so they are roughly twice that size to store in memory.